### ISSUE TITLE
Was it possible to use SPI instead of bitbanging ?

### ISSUE THREAD

**Issue by rogerclarkmelbourne** [engineertype/MAX31856#2]
I've been trying to port this to a ARM based Arduino board, and the code didnt initially work because there wasnt any delays in the bitbanged clocking.

So I tried to re-write using SPI, however i got strange results, and I think it was dropping some bits in between the multibyte transfers.

e.g. The temperature value seemed to be shifted left by 1 bit, and was reading half of what it should be .

So I ended up adding delayMicroseconds(1); between each clock toggle, in your code, to slow down the clock to 500Khz, as this was the only cross platform way I could get to work.

I know this is a relatively slow transfer, but without it, my 120Mhz zero wait state ARM board (GD32F103C8) was running the clock too fast, unless I put in some sort of delay.

I may still try porting to SPI, but as I really just want to get on with my project, converting to use SPI is not my top priority at the moment.

**Comment by engineertype** [comment]
Sorry for the problem's you experienced.  Sure, it is possible to use SPI.  I tried to make this portable, and Arduino's SPI library hides things that would make it more difficult to port to another platform, like Raspberry Pi or BeagleBone.  Adding a delay was the correct solution for you.

Peter

**Comment by rogerclarkmelbourne** [comment]
Hi Peter

No problem

If I get time I may try to get it working with SPI again, in which case I may do a PR, but I suspect I will probably just use the delay solution.
Thanks

Roegr

**Comment by rogerclarkmelbourne** [comment]
Peter.
Just a final note. I found some code that does work with SPI.

It looks like the issue is that a small delay is needed between CS going low and the read / write command, and also a delay between the read register command and the actual reading of the data.

I had tried the delay between CS and the read register command, but I'd not put in the delay between the read register and the transferring of data.

The latter may have been why my SPI implementation was dropping 1 bit of data

It also explains why I couldnt simply setup a buffer with the read command value and 0,0,0,0 and then use the SPI transfer buffer command, as its simply too fast for this device.
